pkg/router/controller/host_admitter_test.go

```diff
@@ -0,0 +1,399 @@
+package controller
```

Comment:
Consider simplifying these tests by validating each function exhaustively in isolation but minimally in combination.  This would allow for fewer test cases to provide more coverage.  At the least I suggest testing addRoute and HandleRoute separately.